lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/gateway/ThirdPartyDocumentGateway.java

```diff
@@ -207,38 +202,59 @@ private DocumentCheckResult responseHandler(CloseableHttpResponse httpResponse)
             throws IOException, ParseException, JOSEException, CertificateException,
                     OAuthHttpResponseExceptionWithErrorBody {
         int statusCode = httpResponse.getStatusLine().getStatusCode();
-        LOGGER.info("Third party response code {}", statusCode);
 
         HttpEntity entity = httpResponse.getEntity();
         String responseBody = EntityUtils.toString(entity);
-        LOGGER.info("Third party response {}", responseBody);
 
         if (statusCode == 200) {
-            DcsResponse unwrappedDcsResponse =
-                    dcsCryptographyService.unwrapDcsResponse(responseBody);
-            validateDcsResponse(unwrappedDcsResponse);
+            LOGGER.info("Third party response code {}", statusCode);
+
+            try {
+                DcsResponse unwrappedDcsResponse =
+                        dcsCryptographyService.unwrapDcsResponse(responseBody);
+                validateDcsResponse(unwrappedDcsResponse);
+
+                DocumentCheckResult documentCheckResult = new DocumentCheckResult();
+                documentCheckResult.setExecutedSuccessfully(true);
+                documentCheckResult.setTransactionId(unwrappedDcsResponse.getRequestId());
+                documentCheckResult.setValid(unwrappedDcsResponse.isValid());
+
+                return documentCheckResult;
+            } catch (IpvCryptoException e) {
+                // Seen when a signing cert has expired and all message signatures fail verification
+                // We need to log this specific error message from the IpvCryptoException for
+                // context
+                LOGGER.error(e.getMessage());
```

Comment:
For logging exceptions were better doing LOGGER.error("Some error reason", e) So that exception stack gets logged

Reply:
IpvCryptoException is thrown in the DcsCryptographyService when unwrapDcsResponse is called but fails. It has two possible values which indicates which part of the unwrap process failed signature verification - expected when a cert expired.

DCS Response Outer Signature invalid.
DCS Response Inner Signature invalid.

I've kept the message as the reason and added the exception to the logger.